docs/platform_developers/developer_getting_started.rst

```diff
@@ -0,0 +1,151 @@
+* Set project sources
+* Module developers guide
+
+
+Setup Inmanta pip artifacts
+#############################
+
+In order to use Inmanta's internal pip artifacts:
+
+.. code-block:: bash
+
+    mkdir   ~/.config/pip
+
+    nano ~/.config/pip/pip.conf
+
+And below text into your ``pip.conf`` file:
+
+::
+    [global]
+    index_url=https://artifacts.internal.inmanta.com/inmanta/dev
+
+
+Verify that you already setup GitLab SSH keys
+###############################################
+
+The steps to add SSH keys have already been mentioned in ``Inmanta Getting Started Guide`` which was emailed to you.
+You can click `here <https://internal.inmanta.com/topics/starting.html>`_ to read it again.
+
+In order to verify that it is working correctly, run below command:
+
+.. code-block:: bash
+    
+    ssh git@git.inmanta.com
+
+
+You should see your name in the response.
+
+
+Install VS Code and Inmanta extension
+#######################################
+
+This step does not need any additional information but if you need some help, you can always ask a colleague.
+
+
+Setting up Python virtual environments
+########################################
+
+Python virtual environments have to be created using ``venv`` by running:
+
+.. code-block:: bash
+    
+    python3 -m venv ~/.virtualenvs/<env_name>
+
+Then activated by running:
+
+.. code-block:: bash
+    
+    source ~/.virtualenvs/<env_name>/bin/activate
+
+**Upgrading your ``pip`` will save you a lot of time and troubleshooting.** you can do so by running:
+
+.. code-block:: bash
+    
+    pip install --upgrade pip
+
+Usually the projects that you will work on, come with ``requirements.txt`` or ``requirements.dev.txt`` to install the required modules, if not, it is a good idea to ``pip install``:
+
+1. inmanta-core
+2. pytest
+3. pytest-inmanta
+4. pytest-inmanta-yang
+
+
+Benefit from linting and code navigation by setting up a project
+##################################################################
+
+At the time of this writing, linting and code navigation in IDEs work only if you have a project, so if you want to work on a single module, you still need to have a project.
+
+There are two things that can be done:
+
+1. Either there is already a project and you ``git clone`` your module there.
+2. Make a project.
+
+Steps to create a project are mentioned `here <https://docs.inmanta.com/community/latest/model_developers/configurationmodel.html>`_ for further reading.
+
+Once you are done with creating a project, you can ``cd`` into that directory and open vs code by running:
+
+.. code-block:: bash
+    
+    code .
+
+Upon opening your vs code, and the ``main.cf`` file, you should see modules downloading in ``libs`` directory.
+
+
+Set project sources
+#####################
+
+The next step is to set the sources of your project so that it knows, where to get its required modules from.
+
+1. Locate the project/module that you want to work on from Gitlab.
```

Comment:
no relevant for most people, for internal docs only.

Reply:
```suggestion
1. Find the module you want to work on
```